carbonmark-api/src/utils/crypto.utils.ts

```diff
@@ -25,3 +20,24 @@ export const formatUSDC = (n: bigint | string): string => {
   const remainderString = remainder.toString().padStart(USDC_DECIMALS, "0"); // 10n -> "000010"
   return `${quotient}.${remainderString}`;
 };
+
+/**
+ * Returns true if the message signer is the targetWallet
+ * Uses ECDSA under the hood to recover the signerWalletAddress
+ */
+export const verifyProfileSignature = (params: {
+  nonce?: number;
+  signature: string;
+  expectedAddress: string;
+}) => {
+  if (!params.signature) return false;
+  // Backwards-compat: nonce may be undefined, append empty string
+  const expectedMessage = SIGN_PROFILE_MESSAGE + `${params?.nonce}`;
```

Comment:
Wouldn't this append `null` or `undefined`?
```suggestion
  // Backwards-compat: nonce may be undefined, append empty string
  const expectedMessage = SIGN_PROFILE_MESSAGE + params?.nonce ?? "";
```

Reply:
You are right! Will add a test and fix this